### PR TITLE
ON-102 Crud voice api callflow

### PR DESCRIPTION
**Pull request by deeborania**
Crud voice api callflow

### CODE REVIEW

File: spec/call_flow_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          :destination => "31612345678",
+        },
+      },
+    ]
+    record = true
+    default = false
+
+    expect(voice_client).to receive(:request)
+                              .with(:post, "call-flows", { :title => title, :steps => steps, :default => default, :record => record })
+                              .and_return('{ "data": [ { "id": "' + call_flow_id + '", "title": "' + title + '", "steps": [ { "id": "' + step_id + '", "action": "transfer", "options": { "destination": "31612345678" } } ], "record": true, "default": false, "createdAt": "2019-11-07T12:37:59Z", "updatedAt": "2019-11-07T12:37:59Z" } ], "_links": { "self": "/call-flows/' + call_flow_id + '" } }')
```
Comment (Funfun) [inline spec/call_flow_spec.rb]: in return statement: you can use `to_json` to make this json more readable.

File: spec/call_flow_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    expect(call_flow.id).to eq call_flow_id
+  end
+
+  it "lists call flows" do
+    voice_client = double(MessageBird::VoiceClient)
+    client = MessageBird::Client.new("", nil, nil, voice_client)
+
+    expect(voice_client).to receive(:request)
+                              .with(:get, "call-flows?perPage=2&page=1", {})
+                              .and_return('{ "data": [ { "id": "cfid-0000", "title": "another title", "steps": [ { "id": "sf001", "action": "transfer", "options": { "destination": "31612345678" } } ], "record": true, "default": false, "createdAt": "2019-11-07T12:37:59Z", "updatedAt": "2019-11-07T12:37:59Z" }, { "id": "cfid-0001", "title": "the title", "steps": [ { "id": "s000000000", "action": "transfer", "options": { "destination": "31612345678" } } ], "record": true, "default": false, "createdAt": "2019-11-07T12:37:59Z", "updatedAt": "2019-11-07T12:37:59Z" } ], "_links": { "self": "/call-flows/cfid-0001" } }')
```
Comment (Funfun) [inline spec/call_flow_spec.rb]: in return statement: you can use `to_json` to make this json more readable.

File: spec/call_flow_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    expect(list.items[0].id).to eq "cfid-0000"
+  end
+
+  it "view a call flow" do
+    voice_client = double(MessageBird::VoiceClient)
+    client = MessageBird::Client.new("", nil, nil, voice_client)
+
+    expect(voice_client).to receive(:request)
+                              .with(:get, "call-flows/cfid-1", {})
+                              .and_return('{ "data": [ { "id": "cfid-1", "title": "the title", "steps": [ { "id": "sfid-1", "action": "transfer", "options": { "destination": "31612345678" } } ], "record": true, "default": false, "createdAt": "2019-11-07T12:37:59Z", "updatedAt": "2019-11-07T12:37:59Z" } ], "_links": { "self": "/call-flows/cfid-1" } }')
```
Comment (Funfun) [inline spec/call_flow_spec.rb]: in return statement: you can use `to_json` to make this json more readable.

File: spec/call_flow_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
+describe "CallFlow" do
```
Comment (Funfun) [inline spec/call_flow_spec.rb]: please format the style of test according http://www.betterspecs.org/ 

at least, define variables using `let`

File: lib/messagebird/callflow.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
+require "messagebird/base"
+
+module MessageBird
+  class CallFlow < MessageBird::Base
+    attr_accessor :id, :title, :record, :steps, :default, :createdAt, :updatedAt
```
Comment (Funfun) [inline lib/messagebird/callflow.rb]: please specify `attr_reader` for those attributes where writer is already defined in the class.

**Comment by Funfun** [comment]
Please update styles according ruby style guides.

File: lib/messagebird/voice_client.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    BASE_ENDPOINT = "https://voice.messagebird.com"
+
+    def initialize(access_key)
+      super(access_key)
+      @endpoint = BASE_ENDPOINT
+    end
+
+    def endpoint()
+      @endpoint
```
Comment (Funfun) [inline lib/messagebird/voice_client.rb]: i see in the project is used constant `https://github.com/messagebird/ruby-rest-api/blob/master/lib/messagebird/http_client.rb#L20`

**Comment by Funfun** [comment]
@deeborania please update your branch

**Comment by Funfun** [comment]
outdated branch, please update with recent master.

**Comment by deeborania** [comment]
All style changes done.

**Pull request merged**